lib/io/hyped_can/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+#![no_std]
+
+/// CAN errors that can occur
+/// From: https://docs.embassy.dev/embassy-stm32/git/stm32f767zi/can/enums/enum.BusError.html,
+/// https://docs.embassy.dev/embassy-stm32/git/stm32f767zi/can/enum.TryWriteError.html
+/// https://docs.embassy.dev/embassy-stm32/git/stm32f767zi/can/enum.TryReadError.html,
+/// and https://docs.embassy.dev/embassy-stm32/git/stm32f767zi/can/enums/enum.FrameCreateError.html
+#[derive(Debug)]
+pub enum CanError {
+    Stuff,
+    Form,
+    Acknowledge,
+    BitRecessive,
+    BitDominant,
+    Crc,
+    Software,
+    BusOff,
+    BusPassive,
+    BusWarning,
+    Full,
+    Empty,
+    Unknown,
+    NotEnoughData,
+    InvalidDataLength,
+    InvalidCanId,
+}
+
+#[derive(Clone)]
```

Comment:
i think this could be `Copy` also if you want, avoids needing to use `.clone()` cos it's simple types that can be memcopied